app/utils/AccountUtils.scala

```diff
@@ -17,18 +17,26 @@
 package utils
 
 import controllers.auth.AuthenticatedRequest
-import models.ActingAsAttorneyFor
+import models.{ActingAsAttorneyFor, AgentToken}
 import play.api.i18n.Messages
 import play.api.mvc.Request
 import uk.gov.hmrc.domain.{SaUtr, TaxIdentifier, Uar}
 
 trait AccountUtils {
-  def getAccount(request: AuthenticatedRequest[_]): TaxIdentifier =
-    request.agentRef.getOrElse(request.saUtr.getOrElse(SaUtr("")))
+  def getAccount(request: AuthenticatedRequest[_]): Option[TaxIdentifier] =
```

Comment:
missing tests

Reply:
AuthorityUtilsSpec:115 will cover this .getAccount None Scenario and there a few other tests in there aswell